### PR TITLE
ENH: LookyLoo expert bot for automated website screenshots

### PR DESCRIPTION
**Pull request by waldbauer-certat**

### CODE REVIEW

**Review by wagner-certat — CHANGES_REQUESTED** [review]
That fixes #1048, please add it to the commit message

**Review by wagner-certat — CHANGES_REQUESTED** [review]
Please add documentation to `docs/user/bots.rst`.

File: intelmq/bots/experts/lookyloo/expert.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class LookyLooExpertBot(Bot):
+    """ LookyLoo expert bot for automated website screenshots """
+    instance_url: str = "http://localhost:5100/"
+
+    __instance: None
+
+    def init(self):
+        if pylookyloo is None:
+            raise MissingDependencyError("pylookyloo, version='>=0.6'")
```
Comment (wagner-certat) [inline intelmq/bots/experts/lookyloo/expert.py:20]:
```suggestion
            raise MissingDependencyError("pylookyloo", version=">=0.6")
```

File: intelmq/bots/experts/lookyloo/expert.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        self.__instance = pylookyloo.Lookyloo(self.instance_url)
+
+    def process(self):
+        event = self.receive_message()
+        url = self.__instance.enqueue(url=event.get('source.url'), user_agent=self.http_user_agent)
+        event.add('screenshot_url', url)
```
Comment (wagner-certat) [inline intelmq/bots/experts/lookyloo/expert.py:27]:
```suggestion
        if 'source.url' in event:
            url = self.__instance.enqueue(url=event.get('source.url'), user_agent=self.http_user_agent)
            event.add('screenshot_url', url)
```

File: docs/user/bots.rst [hunk 3]
```diff
@@ -2093,7 +2093,19 @@ Documentation about IDEA: https://idea.cesnet.cz/en/index
 
 * `test_mode`: add `Test` category to mark all outgoing IDEA events as informal (meant to simplify setting up and debugging new IDEA producers) (default: `true`)
 
+LookyLoo
```
Comment (wagner-certat) [inline docs/user/bots.rst:2096]:
```suggestion
Lookyloo
```

File: docs/user/bots.rst [hunk 4]
```diff
@@ -2093,7 +2093,19 @@ Documentation about IDEA: https://idea.cesnet.cz/en/index
...
+^^^^^^^^
+
+Sends an request to a LookyLoo instance & saves the website screenshot link into `screenshot_url`.
```
Comment (wagner-certat) [inline docs/user/bots.rst:2099]:
```suggestion
Lookyloo is a website screenshotting and analysis tool. For more information and installation instructions visit https://www.lookyloo.eu/

The bot sends a request for `source.url` to the configured Lookyloo instance and saves the retrieved website screenshot link in the field `screenshot_url`. Lookyloo only *queues* the website for screenshotting, therefore the screenshot may not be directly ready after the bot requested it.

The `pylookyloo` library is required for this bot.
The `http_user_agent` parameter is passed on, but not other HTTP-related parameter like proxies.

Events without `source.url` are ignored.
```

**Review by wagner-certat — CHANGES_REQUESTED** [review]

**Review by wagner-certat — APPROVED** [review]

**Review by schacht-certat — APPROVED** [review]

**Pull request merged**